scripts/install.sh

```
@@ -2,123 +1,176 @@
+    check_command grep
+}
+
+# Detect system information
+detect_system() {
+    print_step "Detecting system information..."
+    
+    # Detect OS
+    OS=$(uname -s | tr '[:upper:]' '[:lower:]')
+    
+    # Detect architecture and normalize names
+    local arch=$(uname -m)
```

Shellcheck

[warning] Declare and assign separately to avoid masking return values.

(SC2155)